### PR TITLE
WNMGDS-1944 - Create COMPONENT_MATURITY.md

### PR DESCRIPTION
**Pull request by babsdenney**
###  [WNMGDS-1944](https://jira.cms.gov/browse/WNMGDS-1944)

Here are the instructions the component maturity list and the original content this document was created from. Some of the content in the google doc was changed when imported into this MD file so it isn't 1:1.
- Removed the 'about' sections
- Added a section for zooming to 400% zoom level in the responsive section
- Accepted all recommend edits  from reviewers

### Original documents
See original [Maturity checklist doc](https://docs.google.com/document/d/17ez-kUfsptvJq0Wi3uJbtusTbOLLC8Ko--h8Z7cOwz4/edit?usp=sharing)
See [Maturity model mural discovery board here](https://app.mural.co/t/cmsoc4625/m/cmsoc4625/1615907781042/936d8a8cb21252a0af7fbfbe06aea4f24883b394?sender=ucf13c21d112482a4c5766770)

### CODE REVIEW

File: COMPONENT_MATURITY.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+### Supports Forced Colors Mode (FCM)
+While using FCM the components text is legible and improves readability.
+
+#### About
+In Forced Color Mode (FCM) in Windows, verify in Edge, Chrome, and Firefox that:
+- All background colors are reset to a single color.
+- All elements that need a border have one.
+- All SVG elements and icons have a contrast color applied.
+- Color is not the only means used to convey meaning with elements.
+- Focus styles are clearly indicated.
```
Comment (pwolfert) [inline COMPONENT_MATURITY.md:28]: I like these. They're really clear

File: COMPONENT_MATURITY.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+Includes all Sketch symbols for defined options.
+
+#### About
+CMS uses [Sketch (external link)](https://www.sketch.com/ "Sketch app") as a tool to create designs.
+
+#### When to define as 'Not applicable'?
+When the component is a utility with no visual element.
+
+#### How to test?
+Verify that the responsive code component matches the responsive symbols in Sketch.
```
Comment (line47) [inline COMPONENT_MATURITY.md:138]: This how to test section seems to be for the responsive section below? Should this be updated to `Verify with code that all props and options that can be used in code have an equal option in Sketch.`

File: COMPONENT_MATURITY.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+Components are tested in the following screen readers: 
+- Mac OS
+-- Safari + VoiceOver
+- Windows
+-- NVDA + Edge
+-- JAWS + Edge
+-- iOS + Safari + Mobile VoiceOver (as time permits)
+
+#### When to define as 'Not applicable'?
+When the component has no interactive elements like a button or text input a 'Not applicable' state can be applied.
```
Comment (pwolfert) [inline COMPONENT_MATURITY.md:68]: I actually wonder if this leaves out a lot of components that we do want tested for how they appear to a screen reader. One example would be the `Alert`. It's not interactive, but it has an `aria-labelledby` attribute and some screen-reader-only text (prefixes to headings like "Warning: ") and optional auto-focusing behavior. Even without the auto-focusing behavior, I would want to know that the Alert experience is good for screen readers.

File: COMPONENT_MATURITY.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+View in different responsive sizes with [Storybook controls (internal link)](https://design.cms.gov/storybook/ "CMSDS Storybook") or with browser dev tools.
+
+### Spanish translations
+Includes Spanish translations for default text content.
+
+#### About
+[Read more about Internationalization (internal link)](https://design.cms.gov/getting-started/developers/internationalization/?theme=core "Internationalization") in the CMSDS.
+
+#### When to define as 'Not applicable'?
+When the component has not text to apply a translation to.
```
Comment (pwolfert) [inline COMPONENT_MATURITY.md:121]:
```suggestion
When the component has no text to apply a translation to.
```

File: COMPONENT_MATURITY.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+#### About
+[Storybook (external link)](https://storybook.js.org/ "Storybook JS") is an open-source tool for building components in isolation that is public to all Design System users. 
+
+#### When to define as 'Not applicable'?
+Should not be used.
+
+#### How to test?
+Manually interact with the Storybook stories and make sure that all defined props are in some way tested by our storybook story and that the meaningful ones can be changed. Specifically,
+- Event handlers have visible output through Actions or logging
+- Visual and content props can be changed
```
Comment (pwolfert) [inline COMPONENT_MATURITY.md:99]: I'm working on improving coverage in this area on our stories, but I would like to qualify this

```suggestion
- Visual and content props can be changed (for props that can be easily changed with standard Storybook controls)
```

File: COMPONENT_MATURITY.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+#### When to define as 'Not applicable'?
+Should not be used.
+
+#### How to test?
+Run `yarn test:a11y`
+
+### Supports Forced Colors Mode (FCM)
+While using FCM the components text is legible and improves readability.
+
+#### About
```
Comment (babsdenney) [inline COMPONENT_MATURITY.md:22]: Looking at the 'About' sections, I think most of the about sections are related to "how to test" instructions. How would everyone feel if I got rid of the about section and moved that contents to the "how to test" section? @line47 @pwolfert

File: COMPONENT_MATURITY.md (hunk 3 above)
Comment (babsdenney) [reply COMPONENT_MATURITY.md:68]: This is a great point! There are alot of elements this situation could apply to. @SparrowSpit Would there be any instance where Voiceover testing wouldn't be applicable to the component?

File: COMPONENT_MATURITY.md (hunk 2 above)
Comment (babsdenney) [reply COMPONENT_MATURITY.md:138]: Good catch! Thank you, I'll get this updated.

File: COMPONENT_MATURITY.md (hunk 6 above)
Comment (pwolfert) [reply COMPONENT_MATURITY.md:22]: Yeah, that's a good idea.

Comment (brentschneider) [reply COMPONENT_MATURITY.md:22]: Ditto 👍🏼

**Review by brentschneider — APPROVED** [review]
These edits are in a good spot to release as an MVP candidate.

**Review by pwolfert — APPROVED** [review]
LGTM. I restarted the CI job because it failed during dependency installation (unrelated to your changes)

**Pull request merged**